src/shared/ui/list/list.test.tsx

```diff
@@ -0,0 +1,57 @@
+import { render, screen } from '@testing-library/react';
+import { List, cx } from './list';
+
+const data = ['listItem 1', 'listItem 2', 'listItem 3'];
+
+describe('List Component', () => {
+  it('renders list correctly', () => {
+    render(<List data={data} />);
+
+    const list = screen.getByTestId('list');
+
+    expect(list).toBeVisible();
+  });
+
+  it('list has marked by default', () => {
+    render(<List data={data} />);
+
+    const list = screen.getByTestId('list');
+
+    expect(list).toHaveClass(cx('marked'));
+  });
+
+  it('renders list correctly if list-data is empty', () => {
+    render(<List data={[]} />);
+
+    const list = screen.getByTestId('list');
+
+    expect(list).toBeInTheDocument();
+  });
```

Comment:
I'd add:
```
 const items = screen.getAllByTestId('list-item');

 expect(items).toHaveLength(0);
```

Reply:
Done

Reply:
Done

Reply:
I'd prefer to have this test, but ok, let it be up to you